libs/git_wrapper/Git_wrapper.ml

```diff
@@ -546,7 +546,11 @@ let commit ?cwd msg =
 let get_project_url ?cwd () : string option =
   let cmd = (git, cd cwd @ [ "ls-remote"; "--get-url" ]) in
   match UCmd.string_of_run ~trim:true cmd with
-  | Ok (url, _) -> Some url
+  | Ok (url, _) ->
+      Some
```

Comment:
@brandonspark can you check if osemgrep cli_publish will also do the work?
Also in OCaml we should use the Uri parsing library, which may have some helper function to do this credentials skipping. If you don't have time @brandonspark let me know and I'll investigate using Uri instead of those regexps.

Reply:
I verified that `semgrep publish test.yaml --experimental` will cause a hard-coded project URL `"https://gitlab-ci-token:glcbt-64_wFuiRFQk9t841JHKQnAT@gitlab.company.world/app/test-case.git"` to be replaced with `https://gitlab.company.world/app/test-case.git`. Looks good to me.

Reply:
This code:
```
      let uri = Uri.of_string "https://gitlab-ci-token:glcbt-64_wFuiRFQk9t841JHKQnAT@gitlab.company.world/app/test-case.git" in
      (match Uri.user uri, Uri.password uri with
      | Some a, Some b ->
          UCommon.(pr2 (spf "%s and %s" a b))
      | _ -> ()
        );
 ```
 seems to produce 
 `gitlab-ci-token and glcbt-64_wFuiRFQk9t841JHKQnAT`
 so it looks like URI parsing works better.